iroh-gateway/src/core.rs

```diff
@@ -222,16 +201,16 @@ async fn get_ipfs(
         &HEADER_X_IPFS_PATH,
         HeaderValue::from_str(&full_content_path).unwrap(),
     );
-    // todo(arqu): add X-Ipfs-Roots
 
     // handle request and fetch data
     let req = Request {
         format,
-        cid: CidOrDomain::Cid(cid),
-        full_content_path,
+        cid: resolved_path.root().clone(),
+        resolved_path,
         query_file_name,
-        content_path: cpath.to_string(),
+        content_path: full_content_path.to_string(),
```

Comment:
maybe store the `Path` directly instead of the stringified version?

Reply:
Some of the smaller manipulations require details from the explicit path that was passed in. Some of those details get lost on the `resolver::Path` like whether the path had a trailing `/` at the end or not. Opening an issue as most of them are able to utilize it properly though.